src/dotnet/commands/dotnet-build/BuildCommandParser.cs

```diff
@@ -18,25 +19,42 @@ internal static class BuildCommandParser
                       .With(name: CommonLocalizableStrings.CmdProjectFile,
                             description:
                             "The MSBuild project file to build. If a project file is not specified, MSBuild searches the current working directory for a file that has a file extension that ends in `proj` and uses that file."),
-                CommonOptions.HelpOption(),
-                Create.Option(
-                    "-o|--output",
-                    LocalizableStrings.OutputOptionDescription,
-                    Accept.ExactlyOneArgument()
-                          .With(name: LocalizableStrings.OutputOptionName)
-                          .ForwardAsSingle(o => $"/p:OutputPath={o.Arguments.Single()}")),
-                CommonOptions.FrameworkOption(),
-                CommonOptions.RuntimeOption(),
-                CommonOptions.ConfigurationOption(),
-                CommonOptions.VersionSuffixOption(),
-                Create.Option(
-                    "--no-incremental",
-                    LocalizableStrings.NoIncrementialOptionDescription),
-                Create.Option(
-                    "--no-dependencies",
-                    LocalizableStrings.NoDependenciesOptionDescription,
-                    Accept.NoArguments()
-                          .ForwardAs("/p:BuildProjectReferences=false")),
-                CommonOptions.VerbosityOption());
+                FullBuildOptions
+                );
+
+        private static Option[] FullBuildOptions
+        {
+            get
+            {
+                var fullBuildOptions = new List<Option>
+                {
+                    CommonOptions.HelpOption(),
+                    Create.Option(
+                        "-o|--output",
+                        LocalizableStrings.OutputOptionDescription,
+                        Accept.ExactlyOneArgument()
+                              .With(name: LocalizableStrings.OutputOptionName)
+                              .ForwardAsSingle(o => $"/p:OutputPath={o.Arguments.Single()}")),
+                    CommonOptions.FrameworkOption(),
+                    CommonOptions.RuntimeOption(),
+                    CommonOptions.ConfigurationOption(),
+                    CommonOptions.VersionSuffixOption(),
+                    Create.Option(
+                        "--no-incremental",
+                        LocalizableStrings.NoIncrementialOptionDescription),
+                    Create.Option(
+                        "--no-dependencies",
+                        LocalizableStrings.NoDependenciesOptionDescription,
+                        Accept.NoArguments()
+                              .ForwardAs("/p:BuildProjectReferences=false")),
+                    CommonOptions.NoRestoreOption(),
+                    CommonOptions.VerbosityOption()
+                };
+
+                RestoreCommandParser.AddImplicitRestoreOptions(fullBuildOptions);
```

Comment:
@jonsequitur Do you have any thoughts on how to do this while keeping things fluid?